### PR TITLE
Added new data source `azurerm_eventhub_namespace_authorization_rule`

### LINKED ISSUE #3577: Add data source for azurerm_eventhub_namespace_authorization_rule
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Summary
Please add a new data source for an existing event hub namespace auth rule, in addition to the existing resource that creates authorisation rule resources.

#### Motivation

If using event hubs for diagnostic logging then many resources will need auth rule keys to be able send logs to the hub. Event hubs are limited to 12 auth rules, so an auth rule needs to be reused across many resources diagnostic settings. When using the azurerm_monitor_diagnostic_setting we need to pass the ID of the event hub auth rule in the eventhub_authorization_rule_id  argument, and there is no way to acquire the ID of the rule without creating it in the same template or string building based on the eventhub namespace datasource.

### New or Affected Resource(s)

* azurerm_eventhub_namespace_authorization_rule

### Potential Terraform Configuration

```hcl
data "azurerm_eventhub_namespace_authorization_rule" "test" {
  name                           = "test-rule"
  namespace_name        = "eventhub-ns"
  resource_group_name = "eventhug-rg"
}
```

### PR DESCRIPTION
**Pull request by tracypholmes**
Fixes #3577.
image

### CODE REVIEW

File: azurerm/internal/services/eventhub/data_source_eventhub_namespace_authorization_rule.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: azure.EventHubAuthorizationRuleSchemaFrom(map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
```
[the remark this answers is not in the document]
Comment (tracypholmes) [reply azurerm/internal/services/eventhub/data_source_eventhub_namespace_authorization_rule.go]: Oooh. Agreed.

File: azurerm/internal/services/eventhub/tests/data_source_eventhub_namespace_authorization_rule_test.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+resource "azurerm_eventhub_namespace" "test" {
+  name                = "acctest-EHN-%[1]d"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  sku = "Standard"
+}
+
+data "azurerm_eventhub_namespace_authorization_rule" "test" {
```
[the remark this answers is not in the document]
Comment (tracypholmes) [reply azurerm/internal/services/eventhub/tests/data_source_eventhub_namespace_authorization_rule_test.go]: Gotcha & understood. In this case, I think I removed too much stuff.

File: azurerm/internal/services/eventhub/data_source_eventhub_namespace_authorization_rule.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+	keysResp, err := client.ListKeys(ctx, resourceGroup, namespaceName, name)
+	if err != nil {
+		return fmt.Errorf("Error making Read request on Azure EventHub Authorization Rule List Keys %s: %+v", name, err)
+	}
+
+	d.Set("primary_key", keysResp.PrimaryKey)
+	d.Set("secondary_key", keysResp.SecondaryKey)
+	d.Set("primary_connection_string", keysResp.PrimaryConnectionString)
+	d.Set("secondary_connection_string", keysResp.SecondaryConnectionString)
```
Comment (katbyte) [inline azurerm/internal/services/eventhub/data_source_eventhub_namespace_authorization_rule.go]: We'll need to add this into the schema:

```
Test Failed

------- Stdout: -------
=== RUN   TestAccDataSourceAzureRMEventHubNamespaceAuthorizationRule_basic
=== PAUSE TestAccDataSourceAzureRMEventHubNamespaceAuthorizationRule_basic
=== CONT  TestAccDataSourceAzureRMEventHubNamespaceAuthorizationRule_basic
panic: Invalid address to set: []string{"manage"}
g
```

File: azurerm/internal/services/eventhub/tests/data_source_eventhub_namespace_authorization_rule_test.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+				Config: testAccDataSourceEventHubNamespaceAuthorizationRule_basic(data, true, true, true),
+				Check:  resource.ComposeTestCheckFunc(),
+			},
+		},
+	})
+}
+
+func testAccDataSourceEventHubNamespaceAuthorizationRule_basic(data acceptance.TestData, listen, send, manage bool) string {
+	return fmt.Sprintf(`
+data "azurerm_eventhub_namespace_authorization_rule" "test" {
```
Comment (katbyte) [inline azurerm/internal/services/eventhub/tests/data_source_eventhub_namespace_authorization_rule_test.go]: Could we put this at the bottom of the test block

File: website/docs/d/eventhub_namespace_authorization_rule.html.markdown [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+# Data Source: azurerm_eventhub_namespace_authorization_rule
+
+Use this data source to access information about an Authorization Rule for an Event Hub Namespace.
+
+## Example Usage
+
+```hcl
+data "azurerm_eventhub_namespace_authorization_rule" "example" {
+  name                = "navi"
+  resource_group_name = "${azurerm_eventhub_authorization_rule.example.resource_group_name}"
```
Comment (katbyte) [inline website/docs/d/eventhub_namespace_authorization_rule.html.markdown]:
```suggestion
  resource_group_name = "example-resources"
```

File: azurerm/internal/services/eventhub/data_source_eventhub_namespace_authorization_rule.go [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+func dataSourceEventHubNamespaceAuthorizationRule() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceEventHubNamespaceAuthorizationRuleRead,
+
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: azure.EventHubAuthorizationRuleSchemaFrom(map[string]*schema.Schema{
```
Comment (katbyte) [inline azurerm/internal/services/eventhub/data_source_eventhub_namespace_authorization_rule.go]: THis puts the properties in as non computed, we need to take that schema and place it in here inline as computed istead.

**Comment by alastairtree** [comment]
Thanks for this @tracypholmes

**Pull request merged**